<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigUI</class>
 <widget class="QDialog" name="ConfigUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>513</width>
    <height>377</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../ldforge.qrc">
    <normaloff>:/icons/settings.png</normaloff>:/icons/settings.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabs">
     <property name="tabPosition">
      <enum>QTabWidget::North</enum>
     </property>
     <property name="currentIndex">
      <number>5</number>
     </property>
     <property name="elideMode">
      <enum>Qt::ElideNone</enum>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <property name="tabsClosable">
      <bool>false</bool>
     </property>
     <property name="movable">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Colors</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="whatsThis">
               <string/>
              </property>
              <property name="text">
               <string>Main color:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="mainColorButton">
              <property name="whatsThis">
               <string>This color is used for the main color.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/colorselect.png</normaloff>:/icons/colorselect.png</iconset>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_3">
              <property name="whatsThis">
               <string/>
              </property>
              <property name="text">
               <string>Main color alpha:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSlider" name="mainColorAlpha">
              <property name="whatsThis">
               <string>Opacity of main color in the viewport.</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
              <property name="tickInterval">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_2">
              <property name="whatsThis">
               <string/>
              </property>
              <property name="text">
               <string>Background:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QPushButton" name="backgroundColorButton">
              <property name="whatsThis">
               <string>This is the background color for the viewport.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/colorselect.png</normaloff>:/icons/colorselect.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::ExpandingFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="whatsThis">
            <string/>
           </property>
           <property name="text">
            <string>Line thickness:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QSlider" name="lineThickness">
             <property name="whatsThis">
              <string>How thick lines should be drawn in the viewport.</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksAbove</enum>
             </property>
             <property name="tickInterval">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>#</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="colorizeObjects">
         <property name="whatsThis">
          <string>Makes colored objects (non-16 and 24) appear colored in the list view. A red triangle will, for instance, have its entry written in red text. This can be useful to locate colored objects.</string>
         </property>
         <property name="text">
          <string>Colorize objects in list view</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="colorBFC">
         <property name="whatsThis">
          <string>Polygons' front sides become green and back sides red.</string>
         </property>
         <property name="text">
          <string>Red/green BFC view (incomplete)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="blackEdges">
         <property name="whatsThis">
          <string>Makes all edgelines appear black. If this is not set, edge lines take their color as defined in LDConfig.ldr.</string>
         </property>
         <property name="text">
          <string>Black edges</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="implicitFiles">
         <property name="text">
          <string>List implicitly loaded files</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Shortcuts</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="whatsThis">
          <string>Here you can alter keyboard shortcuts for almost all LDForge actions. Only exceptions are the controls for the viewport. Use the set button to set a key shortcut, clear to remove it and reset to restore the shortcut to its default value.

Shortcut changes apply immediately after closing this window.</string>
         </property>
         <property name="title">
          <string>Shortcuts</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QListWidget" name="shortcutsList">
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAsNeeded</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QPushButton" name="shortcut_set">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="shortcut_reset">
              <property name="text">
               <string>Reset</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/undo.png</normaloff>:/icons/undo.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="shortcut_clear">
              <property name="text">
               <string>Clear</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/delete.png</normaloff>:/icons/delete.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Quick colors</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="whatsThis">
          <string>Here you can alter the layout of the quick colors toolbar. Use the controls to add, remove or edit the colors used. You can also add separators in between colors.

Usually this contains MainColor, EdgeColor and some auxiliary colors used to group objects.</string>
         </property>
         <property name="title">
          <string>Quick colors</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QListWidget" name="quickColorList">
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAsNeeded</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPushButton" name="quickColor_add">
              <property name="text">
               <string>Add Color</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/palette.png</normaloff>:/icons/palette.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_addSep">
              <property name="text">
               <string>Add Separator</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_edit">
              <property name="text">
               <string>Edit</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/mode-draw.png</normaloff>:/icons/mode-draw.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="line_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_moveUp">
              <property name="text">
               <string>Move Up</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/arrow-up.png</normaloff>:/icons/arrow-up.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_moveDown">
              <property name="text">
               <string>Move Down</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/arrow-down.png</normaloff>:/icons/arrow-down.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_remove">
              <property name="text">
               <string>Remove</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/delete.png</normaloff>:/icons/delete.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quickColor_clear">
              <property name="text">
               <string>Clear List</string>
              </property>
              <property name="icon">
               <iconset resource="../../ldforge.qrc">
                <normaloff>:/icons/delete.png</normaloff>:/icons/delete.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Grids</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="grids">
         <property name="title">
          <string>Grids</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>External Programs</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QGroupBox" name="extProgs">
         <property name="whatsThis">
          <string>LDForge supports launching of several third-party utility tools; here you can set the file paths to these tools. Set the paths of the tools to the exe files.

Under Linux, you can also set the programs to be launched with Wine, so you can use Windows binaries here as well. You will obviously need Wine installed. A 'wine' command in PATH is necessary for this to work.</string>
         </property>
         <property name="title">
          <string>External Programs</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>Downloads</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Download path:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="downloadPath"/>
         </item>
         <item>
          <widget class="QPushButton" name="findDownloadPath">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../ldforge.qrc">
             <normaloff>:/icons/folder.png</normaloff>:/icons/folder.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../ldforge.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ConfigUI</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ConfigUI</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
